tests/compiler/inlining.ts

```diff
@@ -6,3 +8,55 @@ export function test(): i32 {
 }
 
 assert(test() == 3);
+
+// Function inlining
```

Comment:
Can we add inlining function test with closure now?

Reply:
In the current state, closures can't be inlined because these compile to indirect function calls. Might be that some static evaluation of indirect to direct calls, where possible, can help, but that wouldn't be straight forward for most users to know beforehand.

Reply:
I mean checking how inlined function that return a closure

Reply:
Should theoretically inline just fine, as `(a: i32): i32 => a` compiles to a constant table index. Going to add a test.

Reply:
Great!